osvr/RenderKit/RenderManagerBase.cpp

```diff
@@ -1704,18 +1737,179 @@ namespace renderkit {
 
             /// Compute the full matrix by multiplying the parts.
             Eigen::Projective3f full =
-                postTranslation * postScale * postProjectionTranslate *
+                postTranslation * justInTimeWarp * postScale * postProjectionTranslate *
                 lastModelViewTransform * currentModelViewInverseTransform *
                 preProjectionTranslate * preScale * preTranslation;
 
-            // Store the result.
+            // Store the result, transposing if we're using D3D.
             matrix16 timeWarp;
-            Eigen::Matrix4f::Map(timeWarp.data) = full.matrix();
+            if (doTranspose) {
+              Eigen::Matrix4f::Map(timeWarp.data) = full.matrix().transpose();
+            } else {
+              Eigen::Matrix4f::Map(timeWarp.data) = full.matrix();
+            }
             m_asynchronousTimeWarps.push_back(timeWarp);
         }
         return true;
     }
 
+    std::array<float, 4> RenderManager::ComputeJustInTimeWarp() {
+      // We initialize the values with ones that won't cause any
+      // change.  We will override them as we find reason.
+      std::array<float, 4> ret = { 1, 1, 0, 0 };
+
+      // Figure out which edge of the display scan-out starts at based
+      // on the just-in-time rotation.  This describes which edge will
+      // be rotated to point "up" when the display is rotated about the
+      // +Z axis (out of the screen) and it starts at the canonical orientation
+      // with X to the right and Y up.  The four results are 0 = top, 1 = right,
+      // 2 = bottom, 3 = left.  The code rounds to the nearest one.
+      int edgeUp = static_cast<int>(
```

Comment:
I know @godbyk has done a fair bit of coding over in steamvr-osvr land relating to scan-out direction...